test/runnable/testcontracts.d

```diff
@@ -698,6 +698,139 @@ void test9383()
+    void bar4(string s) out { A15524.val = s;             } body { }                                // no closure
+    void bar5(string s) out { A15524.val = s;             } body { }                                // no closure
+    void bar6(string s) out { A15524.val = s;             } body { string x; dg = { x = null; }; }  // closure [local]
+    void bar7(string s) out { A15524.val = s;             } body {           dg = { s = null; }; }  // closure [parameter]
+}
+
+class B15524 : A15524
+{
+    static string val;
+
+    // in-contract tests
+    override void foo1(string s) in  { B15524.val = s; } body { }                                   // -> no closure
+    override void foo2(string s) in  { B15524.val = s; } body { string x; dg = { x = null; }; }     // -> closure [local] appears
+    override void foo3(string s) in  { B15524.val = s; } body {           dg = { s = null; }; }     // -> closure [parameter]
+    override void foo4(string s) in  { B15524.val = s; } body { string x; dg = { x = null; }; }     // -> closure [local] appears
+    override void foo5(string s) in  { B15524.val = s; } body {           dg = { s = null; }; }     // -> closure [parameter] appears
+    override void foo6(string s) in  { B15524.val = s; } body { }                                   // -> closure [local] disappears
+    override void foo7(string s) in  { B15524.val = s; } body { }                                   // -> closure [parameter] disappears
+
+    // out-contract tests
+    override void bar1(string s) out { B15524.val = s; } body { }                                   // -> no closure
+    override void bar2(string s) out { B15524.val = s; } body { string x; dg = { x = null; }; }     // -> closure [local] appears
+    override void bar3(string s) out { B15524.val = s; } body {           dg = { s = null; }; }     // -> closure [parameter]
+    override void bar4(string s) out { B15524.val = s; } body { string x; dg = { x = null; }; }     // -> closure [local] appears
+    override void bar5(string s) out { B15524.val = s; } body {           dg = { s = null; }; }     // -> closure [parameter] appears
+    override void bar6(string s) out { B15524.val = s; } body { }                                   // -> closure [local] disappears
+    override void bar7(string s) out { B15524.val = s; } body { }                                   // -> closure [parameter] disappears
+}
+
+void test15524()
+{
+    auto a = new A15524();
+    auto b = new B15524();
+
+    // base class in-contract is used from derived class.           // base                   derived
+    b.foo1("1"); assert(A15524.val == "1" && B15524.val == "1");    // no closure          -> no closure
+    b.foo2("2"); assert(A15524.val == "2" && B15524.val == "2");    // closure [local]     -> closure [local] appears
+    b.foo3("3"); assert(A15524.val == "3" && B15524.val == "3");    // closure [parameter] -> closure [parameter]
```

Comment:
I guess that currently all 64bit platforms has problems with the issue 15524 case - closure allocation + contracts with virtual class methods. This PR works for Win64, but I think other Posix platforms need other fix.

But, I'm mainly working on Windows platform. So I'll pending this PR till finishing other possible regression fixes done.